currency-rates/internal/app/app.go

```diff
@@ -25,6 +25,22 @@ func NewApp(c Config, l *zap.SugaredLogger) *App {
 	}
 }
 
+func (app *App) initCron(ctx context.Context, subscriptionService services.SubscriptionService) gocron.Scheduler {
+	s, _ := gocron.NewScheduler()
+	
+	_, _ = s.NewJob(
+		gocron.CronJob(
+			"0 2 * * *",
+			false,
+		),
+		gocron.NewTask(
+			subscriptionService.NotifySubscribers, ctx,
+		),
+	)
+	
+	return s;
+}
+
 func (app *App) Run(ctx context.Context) error {
```

Comment:
Run command is still violates SRP. You have a lot of constructor calls in Run function, but they must be in New constructor. Your class App should have all the dependencies it need to start an application.

Reply:
Done